src/utils/filter.py

```
@@ -545,28 +545,46 @@
-        "%Y-%m-%dT%H:%M:%S.%fZ",  # 2024-01-01T12:00:00.123456Z
-        "%Y-%m-%d",  # 2024-01-01
+        "%Y-%m-%dT%H:%M:%S.%fZ",  # 2024-01-01T12:00:00.123456Z (UTC)
+        "%Y-%m-%dT%H:%M:%S%z",  # 2024-01-01T12:00:00+00:00
+        "%Y-%m-%dT%H:%M:%S.%f%z",  # 2024-01-01T12:00:00.123456+00:00
     ]
 
-    for fmt in datetime_formats:
+    for fmt in timezone_aware_formats:
         try:
-            return datetime.datetime.strptime(value, fmt)
+            parsed = datetime.datetime.strptime(value, fmt)
```

Ruff (0.12.2)

Naive datetime constructed using `datetime.datetime.strptime()` without %z

(DTZ007)